titus-client/src/main/java/com/netflix/titus/runtime/endpoint/v3/grpc/GrpcJobManagementModelConverters.java

```diff
@@ -406,7 +411,32 @@ private static SharedContainerVolumeSource toCoreSharedVolumeSource(com.netflix.
                 .build();
     }
 
-    public static DisruptionBudget toCoreDisruptionBudget(com.netflix.titus.grpc.protogen.JobDisruptionBudget grpcDisruptionBudget) {
+    private static List<PlatformSidecar> toCorePlatformSidecars(List<com.netflix.titus.grpc.protogen.PlatformSidecar> platformSidecarList) {
+        List<PlatformSidecar> platformSidecars = new ArrayList<>();
+        for (com.netflix.titus.grpc.protogen.PlatformSidecar ps : platformSidecarList) {
+            platformSidecars.add(toCorePlatformSidecar(ps));
+        }
+        return platformSidecars;
+    }
+
+    private static PlatformSidecar toCorePlatformSidecar(com.netflix.titus.grpc.protogen.PlatformSidecar ps) {
+        return PlatformSidecar.newBuilder()
+                .withName(ps.getName())
+                .withChannel(ps.getChannel())
+                .withArguments(structToJSONString(ps.getArguments(), ps.getName()))
+                .build();
+    }
+
+    private static String structToJSONString(Struct arguments, String sidecarName) {
+        try {
+            return JsonFormat.printer().omittingInsignificantWhitespace().print(arguments);
+        } catch (InvalidProtocolBufferException e) {
+            throw new IllegalArgumentException("Unable to serialize arguments for the " + sidecarName + " sidecar: " + e.getMessage());
```

Comment:
We must handle this error somehow in GRPC server and return validation error to the caller just like we do it for other fields. We cannot use here `EntityValidator`, so this must be done explicitly in the GRPC call handler. We do it already for some other fields.

Reply:
Did you mean `IllegalArgumentException` not `EntityValidator`?

This looked like the standard way to throw errors in invalid grpc proto input in this class, so I figured `IllegalArgumentException` would be handled similarly in the upper layers.

Reply:
I meant that ideally validation would be handled by `EntityValidator`, but it works on the internal Job object (not protobuf), so it is not possible to use it that early.
We map validation errors to GRPC model. Please, have a look at the `com.netflix.titus.runtime.endpoint.v3.grp.ErrorResponses` class.

Reply:
Hmm. I'll create circular imports if I try to use that here.
Are you sure I should be using it directly?

What if instead I use the `TitusServiceException` thing? (which in turn uses ErrorResponses to return a sane GRPC error code).
If I use that, the http/grpc error I get looks like:


```
E           tests.titus_client.titus_client_exceptions.HTTPClientException: 400: Error Bad Request for url: https://titusapi-test-devkyleacell001.vip.us-east-1.test.cloud.netflix.net:7004/api/v3/jobs: {'message': 'Error creating core job descriptor: Unable to '
E                       'serialize arguments for the sleepy sidecar: HI',
E            'statusCode': 400}
```

Which I think is what you wanted to accomplish, to fail way early?